lib/src/custom_sentence.dart

```diff
@@ -0,0 +1,22 @@
+import 'package:nmea/src/checksum_sentence.dart';
+import 'package:nmea/src/nmea_sentence.dart';
+import 'package:nmea/src/nmea_sentence_type.dart';
+
+/// The data format is completely customizable, as long as it fits in the general
+/// envelope of an NMEA0183 sentence (begins with '$' and ends with windows-
+/// style newline characters).
+class CustomSentence extends ChecksumSentence {
+  /// Returns the identifier used to identify this sentence type
+  final String identifier;
+  final bool validateChecksums;
+
+  CustomSentence({required this.identifier, required super.raw, this.validateChecksums = true})
+      : super(
+          type: NmeaSentenceType.unknown,
+          prefix: nmeaPrefix,
+        );
+
+
+  @override
+  bool get valid => super.valid || !validateChecksums;
```

Comment:
I wouldn't override the `valid` property, instead you should move this to `hasValidChecksum`:

```dart
  bool get hasValidChecksum => !validateChecksums || super.hasValidChecksum;
```

We still want to validate all the other aspects of the sentence (only ASCII characters, max 80 characters, starts with $ and ends with \r\n, etc.)

Can you also add a test for this case? An invalid custom sentence (missing \r\n for example) is also invalid if the checksum skip is checked.

Reply:
Changed it to the different override. I am not adding another test given this library doesn't have many tests to begin with and this is almost becoming the most tested feature at this point. I originally made these changes for myself and this back and forth is now taking more time than I'd like to spend to get it merged in. Happy for you to make any additional changes that you would want.

Reply:
That's ok! I understand this overhead is much and I don't expect too much. It was a stretch even asking you if you can provide some data (which you did and is awesome!).

Thanks for your work on this!

Reply:
No problem! Happy that I could contribute!